src/main/java/roomescape/reservationTime/dao/H2ReservationTimeDao.java

```diff
@@ -0,0 +1,66 @@
+package roomescape.reservationTime.dao;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.boot.autoconfigure.service.connection.ConnectionDetails;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Component;
+import roomescape.common.Dao;
+import roomescape.reservationTime.ReservationTime;
+
+@Component
+public class H2ReservationTimeDao implements Dao<ReservationTime> {
+    private final JdbcTemplate jdbcTemplate;
+
+    public H2ReservationTimeDao(JdbcTemplate jdbcTemplate, ConnectionDetails connectionDetails) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public ReservationTime add(ReservationTime time) {
+        String sql = "insert into reservation_time(start_at) values (?)";
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+
+        jdbcTemplate.update(connection -> {
+            PreparedStatement preparedStatement = connection.prepareStatement(sql, new String[]{"id"});
+            preparedStatement.setString(1, time.startAt().toString());
+            return preparedStatement;
+        }, keyHolder);
+
+        Long id = keyHolder.getKey().longValue();
+        return new ReservationTime(id, time.startAt());
+    }
+
+    @Override
+    public Optional<ReservationTime> findById(Long id) {
+        String sql = "select id, start_at from reservation_time where id = ?";
+        try {
+            return Optional.ofNullable(jdbcTemplate.queryForObject(sql,
```

Comment:
```suggestion
            return Optional.of(jdbcTemplate.queryForObject(sql,
```
try-catch 로 null이 오는 경우를 막고 있어서 이렇게 사용해도 될 것 같아요 😄

Reply:
오호 그러네요! 감사합니다!